### ISSUE TITLE
Switch to monorepo (using yarn workspaces & lerna)

### ISSUE THREAD

**Issue by dmihal** [opengsn/gsn#452]
One pain point in using this library is that the client-side library includes all the dependencies needed to run a relayer node, such as `express`.

I'd suggest refactoring the repo into a monorepo using Yarn workspaces & Lerna, so that the different modules (client library, relayer node, contracts, etc) can be kept separately.

**Comment by dmihal** [comment]
Just want to circle back on this again, as the number of dependencies of the OpenGSN module causes me a lot of pain. I'll be removing GSN from the core Burner Wallet libraries, as it's just too difficult to deal with.

In total @opengsn/gsn currently has 506 dependencies. That's an insane number for a package that should be very lightweight, and makes it very difficult to use as a client-side library.

**Comment by lirazsiri** [comment]
Hi David, thanks for reporting this and following up. Sorry for the late response! We've added making sure we respond quickly to GitHub issues to our daily checklist. 

We recognize that the issue you're reporting is a dealbreaker for dapp devs and is important enough to block the launch on. We must reduce the relay client's dependencies to something reasonable. Maybe we could get away with a monorepo that pulls in 506 dependencies in alpha stage only people using the repo is the dev team. That is definitely a dealbreaker for actual usage. 

To summarize, before the launch we'll untangle the client's dependencies from all the others as much as possible. After the launch, we will prioritize switching the client to ethers.js to cut back on Web3 dependencies.

**Comment by forshtat** [comment]
Hello David! I hope this message finds you well. 

Better late than never. The migration to 'yarn workspaces' and 'lerna' has recently been merged to master and the next release will be available in two forms: as a single NPM package and as a set of modular packages.

Moving forward we will only support the modular approach, of course.
